config/v1/types_ingress.go

```diff
@@ -151,8 +152,23 @@ type AWSIngressSpec struct {
 	// +kubebuilder:validation:Enum:=NLB;Classic
 	// +kubebuilder:validation:Required
 	Type AWSLBType `json:"type,omitempty"`
+
+	// networkLoadBalancerParameters holds configuration parameters for an AWS
+	// network load balancer. Present only if type is NLB.
+	//
+	// +optional
+	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
 }
 
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network load balancer.
+// +openshift:enable:FeatureGate=SetEIPForNLBIngressController
```

Comment:
this probably more appropriately belongs on either `networkLoadBalancer` above or `eipAllocations` below.  I'd probably choose the `eipAllocations`